### PR TITLE
changes to match EEA V2 spec

### PR DESCRIPTION
**Pull request by macfarla**
* changed IP from bytes to string
* combined enodeHigh + enodeLow to enodeId (single string) 
* don't modify to hex, pass strings as is
* return boolean

Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

### CODE REVIEW

**Comment by chaals** [comment]
These changes match the v2 Permissioned Chains spec, and the v6 Client Spec.

Would it help if EEA brought the spec version numbers in alignment? (e.g. by skipping v3-v6 for the permissioned chains spec?

(Yeah, this is a somewhat "bikeshed" comment...)

File: contracts/ExposedNodeRulesList.sol [hunk 1]
```diff
@@ -5,23 +5,23 @@ import "./NodeRulesList.sol";
 
 contract ExposedNodeRulesList is NodeRulesList {
 
-    function _calculateKey(bytes32 _enodeHigh, bytes32 _enodeLow, bytes16 _ip, uint16 _port) public pure returns(uint256) {
-        return calculateKey(_enodeHigh, _enodeLow, _ip, _port);
+    function _calculateKey(string memory _enodeHigh, string memory _ip, uint16 _port) public pure returns(uint256) {
```
Comment (lucassaldanha) [inline contracts/ExposedNodeRulesList.sol:8]: We should probably rename `_enodeHigh` to `_enode` (applies for all methods in this contract).

File: contracts/NodeIngress.sol [hunk 2]
```diff
@@ -22,29 +22,18 @@ contract NodeIngress is Ingress {
...
-        bytes32 sourceEnodeHigh,
-        bytes32 sourceEnodeLow,
-        bytes16 sourceEnodeIp,
-        uint16 sourceEnodePort,
-        bytes32 destinationEnodeHigh,
-        bytes32 destinationEnodeLow,
-        bytes16 destinationEnodeIp,
-        uint16 destinationEnodePort
-    ) public view returns (bytes32) {
+        string memory sourceEnodeId,
```
Comment (lucassaldanha) [inline contracts/NodeIngress.sol:25]: The `source` prefix doesn't make sense anymore. We should remove it from the parameter names. I think we should use `enodeId`, `ip` and `port`.

File: contracts/NodeRulesProxy.sol [hunk 3]
```diff
@@ -2,13 +2,8 @@ pragma solidity 0.5.9;
 
 interface NodeRulesProxy {
     function connectionAllowed(
```
Comment (lucassaldanha) [inline contracts/NodeRulesProxy.sol:4]: Same as above: remove `source` prefix

File: genesis.json [hunk 4]
```diff
@@ -44,7 +44,7 @@
...
...ffffffffffffffffffffff1660026000600154815260200190815260200160002060009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff161415610e955760019050610f85565b60026000600154815260200190815260200160002060009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1663fe9fbb80836040518263ffffffff1660e01b8152600401808273ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200191505060206040518083038186803b158015610f4757600080fd5b505afa158015610f5b573d6000803e3d6000fd5b505050506040513d6020811015610f7157600080fd5b810190808051906020019092919050505090505b91905056fe4f6e6c792052756c657320636f6e74726163742063616e20747269676765722052756c6573206368616e6765206576656e74734e6f7420617574686f72697a656420746f2075706461746520636f6e74726163742072656769737472792e4d7573742068617665206174206c65617374206f6e65207265676973746572656420636f6e747261637420746f20657865637574652064656c657465206f7065726174696f6e2e436f6e74726163742061646472657373206d757374206e6f74206265207a65726f2ea265627a7a72305820025122701f7a41372cb6e1a346843fd0ea18711295bf39670be2c84addcd881564736f6c63430005090032",
```
Comment (lucassaldanha) [inline genesis.json:47]: I remember that we discussed this before, but I don't remember the decision. Are we updating the docs that are referring to this genesis file (like the tutorials and other stuff)?

File: src/util/enodetools.ts [hunk 5]
```diff
@@ -35,72 +29,31 @@ export const enodeToParams = (enodeURL: string) => {
...
-    enodeLow,
+    enodeId,
     ip,
     port,
     identifier
   };
 };
 
-export const identifierToEnodeHighAndLow = (identifier: string) => {
+export const identifierToEnodeId = (identifier: string) => {
```
Comment (lucassaldanha) [inline src/util/enodetools.ts:53]: I believe we don't need this method anymore, do we? Wasn't it used to "glue" the two enodeId pieces into one?

File: test/test-node-ingress-proxy.js [hunk 6]
```diff
@@ -5,14 +5,14 @@ const Admin = artifacts.require('Admin.sol');
...
-var node1Host = "0x0000000000000000000011119bd359fd";
+var enode1 = "9bd359fdc3a2ed5df436c3d8914b1532740128929892092b7fcb320c1b62f375"
++ "2e1092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929";
+var node1Host = "127.0.0.1";
 var node1Port = 1;
 
-var node2High = "0x892092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929";
-var node2Low = "0xcb320c1b62f37892092b7f59bd359fdc3a2ed5df436c3d8914b1532740128929";
-var node2Host = "0x0000000000000000000011119bd359fd";
+var enode2 = "892092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929"
```
Comment (lucassaldanha) [inline test/test-node-ingress-proxy.js:13]: Are we using `enode2` at all? If we are not, we can delete it and rename `enode1` to `enode` :)

File: test/test-node-ingress.js [hunk 7]
```diff
@@ -5,14 +5,14 @@ const AdminContract = artifacts.require("Admin.sol");
...
-var nodeHost = "0x0000000000000000000011119bd359fd";
+var enode1 = "9bd359fdc3a2ed5df436c3d8914b1532740128929892092b7fcb320c1b62f375"
++ "892092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929";
+var nodeHost = "localhost";
 var nodePort = 30303;
 
-var node2High = "0x892092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929";
-var node2Low = "0xcb320c1b62f37892092b7f59bd359fdc3a2ed5df436c3d8914b1532740128929";
-var node2Host = "0x0000000000000000000011119bd359fd";
+var enode2 = "892092b7fcb320c1b62f3759bd359fdc3a2ed5df436c3d8914b1532740128929"
```
Comment (lucassaldanha) [inline test/test-node-ingress.js:13]: Same: are we using this?

File: src/util/enodetools.ts (hunk 5 above)
Comment (macfarla) [reply src/util/enodetools.ts:53]: it still extracts just the Id part from Id_host_port

File: test/test-node-ingress-proxy.js (hunk 6 above)
Comment (macfarla) [reply test/test-node-ingress-proxy.js]: done

**Review by lucassaldanha — APPROVED** [review]

**Comment by macfarla** [comment]
> These changes match the v2 Permissioned Chains spec, and the v6 Client Spec.
> 
> Would it help if EEA brought the spec version numbers in alignment? (e.g. by skipping v3-v6 for the permissioned chains spec?
> 
> (Yeah, this is a somewhat "bikeshed" comment...)



> These changes match the v2 Permissioned Chains spec, and the v6 Client Spec.
> 
> Would it help if EEA brought the spec version numbers in alignment? (e.g. by skipping v3-v6 for the permissioned chains spec?
> 
> (Yeah, this is a somewhat "bikeshed" comment...)

@chaals we don't have strong opinions on the version number. If they are the same it's easier in a way - but then why are they different things? And also v2 of the Permissioned Chains spec doesn't _exactly_ match v6 of the Client spec (connectionAllowed return type bytes vs bool)

**Pull request merged**